### PR TITLE
Update/available bidders ws integration

### PR DESCRIPTION
**Pull request by scott062**
Depends on: https://github.com/MetaPhase-Consulting/mock-fsbid/pull/166

1. Updates GET call in available bidders view to use client_services in FSBid
- Note: I left the new FSBID GET request with all the other TMAP specific actions in that view (design choice?) 
2. Update client_services to handle new calls
3. Updated common get_results to deal with no pagination function (will probably add in the future) 
- Double check AP and PV pagination doesn't break (relies on the common pagination logic)

### CODE REVIEW

File: talentmap_api/fsbid/services/common.py [hunk 1]
```diff
@@ -203,8 +203,9 @@ def send_get_request(uri, query, query_mapping_function, jwt_token, mapping_func
     '''
     Gets items from FSBid
     '''
+    pagination = get_pagination(query, count_function(query, jwt_token)['count'], base_url, host) if count_function else {}
     return {
-        **get_pagination(query, count_function(query, jwt_token)['count'], base_url, host),
+        **pagination,
         "results": get_results(uri, query, query_mapping_function, jwt_token, mapping_function, api_root)
     }
```
Comment (scott062) [inline talentmap_api/fsbid/services/common.py:210]: Make sure to double check your pagination works for AP/PV - I checked and its good for me

File: talentmap_api/fsbid/services/client.py [hunk 2]
```diff
@@ -488,3 +489,127 @@ def fsbid_assignments_to_tmap(assignments):
...
+    cdo = 'cdo' if isCDO else 'bureau'
+    uri = f"clients/availablebidders/{cdo}"
+    response = send_get_request(
+        uri,
+        query,
+        convert_available_bidder_query,
+        jwt_token,
+        fsbid_available_bidder_to_talentmap,
+        False, # No count function
+        f"/api/v1/client/availablebidders/{isCDO}",
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/client.py:519]: I think you want to change this from `{isCDO}` to `{cdo}`?

File: talentmap_api/fsbid/services/client.py [hunk 3]
```diff
@@ -488,3 +489,127 @@ def fsbid_assignments_to_tmap(assignments):
...
+    assignments = None
+    position = None
+    location = {}
+
+    if employee is not None:
+        current_assignment = employee.get('currentAssignment', None)
+
+    if employee.get('assignment', None) is not None:
+        assignments = employee.get('assignment', None)
+        # handle if array
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/client.py:538]: We can abstract all of this to a common function when we get some time

File: talentmap_api/fsbid/services/client.py [hunk 4]
```diff
@@ -614,9 +614,14 @@ def fsbid_available_bidder_to_talentmap(data):
...
-        "available_bidder_details": data.get("details", None),
+        "available_bidder_details": data.get("details", {}),
     }
-
+    if res['available_bidder_details'] is not None:
+        shared = res['available_bidder_details']['is_shared']
+        archived = res['available_bidder_details']['archived']
+        res['available_bidder_details']['is_shared'] = True if shared == '1' else False
+        res['available_bidder_details']['archived'] = True if archived == '1' else False
+    return res
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/client.py:624]: Take a look at what Elizabeth has done with `pydash get`. Basically lodash for python

Comment (scott062) [reply talentmap_api/fsbid/services/client.py:624]: I'm silly, python has built in get(key, optionalValue)

File: talentmap_api/bureau/services/available_bidders.py [hunk 5]
```diff
@@ -47,28 +49,49 @@ def get_available_bidders_csv(jwt_token, is_cdo):
...
+        "cdo_email": {"path": 'cdo.email', },
     }
 
-    for record in data:
+    for record in data["results"]:
+        languages = f'' if pydash.get(record, ["languages"]) else "None listed"
+        if languages is not "None listed":
+            for language in record["languages"]:
+                languages += f'{language["custom_description"]}, '
+        languages = languages.rstrip(', ')
```
Comment (mjoyce91) [inline talentmap_api/bureau/services/available_bidders.py:79]: never heard of `rstrip` 👍

File: talentmap_api/cdo/services/available_bidders.py [hunk 6]
```diff
@@ -18,7 +19,7 @@
 API_ROOT = settings.FSBID_API_URL
 
 
-def get_available_bidders(jwt_token):
+def get_available_bidders_stats():
```
Comment (elizabeth-jimenez) [inline talentmap_api/cdo/services/available_bidders.py:22]: i'll update the description

File: talentmap_api/bureau/services/available_bidders.py (hunk 5 above)
Comment (scott062) [reply talentmap_api/bureau/services/available_bidders.py:79]: Right, its such a cool find. rstrip and lstrip. Nice @elizabeth-jimenez

File: talentmap_api/bureau/services/available_bidders.py [hunk 7]
```diff
@@ -21,6 +22,7 @@ def get_available_bidders(jwt_token, is_cdo):
     '''
     Returns all clients in Available Bidders list
     '''
+    # deprecated??
```
[the remark this answers is not in the document]
Comment (scott062) [reply talentmap_api/bureau/services/available_bidders.py]: Removed

File: talentmap_api/cdo/services/available_bidders.py [hunk 8]
```diff
@@ -29,39 +30,22 @@ def get_available_bidders(jwt_token):
         'OC': 0,
         'AWOL': 0,
     }
-    if len(ab) > 0:
+    if len(ab):
```
Comment (elizabeth-jimenez) [inline talentmap_api/cdo/services/available_bidders.py:33]: changed from this `if len(ab) > 0:`

File: talentmap_api/cdo/services/available_bidders.py [hunk 9]
```diff
@@ -29,39 +30,22 @@ def get_available_bidders(jwt_token):
...
         results = ab.values('bidder_perdet', 'status', 'oc_reason', 'oc_bureau', 'comments', 'is_shared')
-        # TEMPORARY while we have a better solution merge ab with clients
```
Comment (elizabeth-jimenez) [inline talentmap_api/cdo/services/available_bidders.py:34]: i dont think we are using `results` anymore

**Review by mjoyce91 — APPROVED** [review]

File: talentmap_api/cdo/services/available_bidders.py (hunk 8 above)
Comment (scott062) [reply talentmap_api/cdo/services/available_bidders.py]: changed to  `if ab:`

**Review by elizabeth-jimenez — APPROVED** [review]
🍰 🥮 👍

**Pull request merged**